### ISSUE TITLE
Multiple tasks / same target?

### ISSUE THREAD

**Issue by ysavourel** [GALAglobal/TAPICC-API-implementation#44]
Currently one given asset can be associated with one or more tasks.

But nowhere (I think) we specify that each task needs to be for a different target language.
So it means we can have several tasks for the same target:
- can the be the same type of task? (probably not)
- do they have to be done in a given order or can they be done in parallel?
- if they are to be done sequentially, how do we know the sequence? (relying on the taskId seems dangerous)
- does it even make sense to have several tasks for the same target per asset?

In other words, it seem we need to specify constraints. And i guess it goes back to the need for a prose documentation with all those bits of information.

**Comment by Alino** [comment]
>can the be the same type of task? (probably not)

In the current implementation, it's possible. There is no limitation for this.

>do they have to be done in a given order or can they be done in parallel?

right now, there is no forced order.

>does it even make sense to have several tasks for the same target per asset?

I think yes if the type of the Task is different. Probably some types of Tasks can coexist for the same target language on the same Asset.

Potential Task types:

- Translation,
- Localization,
- Literary editing,
- Technical editing,
- Proofreading,
- Reconciliation,
- Review (also sometimes called “check” or “verification”),
- Review with LQA,
- Terminology extraction,
- Linguistic verification,
- Technical verification,
- Service linguistic task, (here we will put such linguistic tasks as TM management or aligning, for example, as well as manual terminology extraction – whatever is hard to measure productivity or productivity varies)
- MT post-editing,
- Light MT post-editing,
- Creating voice over script from audio/video,
- Retyping handwritten text,
- Copywriting,
- Copyediting,
- Technical writing,
- Adaptation

----

Your questions should be considered by the group, I think you are making a good point especially with the need for correct sequence of the Tasks.
